docs/multitenancy/proposal.md

```diff
@@ -0,0 +1,71 @@
+In a multitenant configuration installing and managing the CRDs and webhooks will be a separate step from deploying operators (and will require cluster admin privileges).
+
+The conversion webhooks in the CRDs and the validating and mutating webhook configurations that are defined for different ASO resources will all point to the `azureoperator-webhook-service` service deployed in the `azureoperator-system` namespace (by default).
+The service will be backed by a deployment of the normal ASO operator running in a restricted webhook-only mode - it won't be watching any namespaces for Azure resources and won't have access to any Azure credentials.
+This mode will be triggered by the configuration setting `AZURE_WEBHOOK_ONLY=1`.
+With this set no other configuration values will be required, and no resource watches will be started - the operator will just respond to webhook requests from the API server.
+
+### RBAC
+
+At the moment installing a global instance of ASO creates a `ClusterRole` defining all of the permissions the operator requires on the different Azure resources (`azureoperator-manager-role`), and then a `ClusterRoleBinding` granting those permissions for resources in all namespaces (`azureoperator-manager-rolebinding`).
+A multitenant deployment won't have the global `ClusterRoleBinding`; instead each managed namespace N will need to contain a `RoleBinding` connecting the global `ClusterRole` with the service account of the operator that will manage N's resources.
+
+### Advantages and constraints
+
+The primary benefit of this architecture is that it enables different tenants to manage their operators independently along with their credentials and resources.
+Additionally in the case of operators using managed identities it removes the need to replicate `AzureIdentity` and `AzureIdentityBinding` resources between namespaces to ensure that the pod identity machinery configures the node identities correctly for the operator.
+
+The biggest downside (other than complexity) is that the different operator installations will be separately managed and we can't assume that they will all be upgraded at the same time.
+The CRDs and webhook service deployment will need to be upgraded by the cluster administrator before any of the tenant operators can be upgraded.
+Conversely if there's a release of the operator that removes an old version of a resource definition, the cluster administrator will need to ensure that all of the tenant operators and resources have been upgraded beyond that version before the core CRDs and webhook service are upgraded.
+The [storage-version-migrator operator](https://github.com/kubernetes-sigs/kube-storage-version-migrator) might be useful for this.
+
+### Alternatives
+
+An earlier plan for implementing multitenancy in ASO was to keep the operator deployment as it is now (one per cluster) but add flexibility in how the credentials would be looked up.
```

Comment:
I think the eventual plan is still to do this, right?

As discussed for other customers such as CAPZ the N copies of operator pattern doesn't work.

Reply:
Oh good point. I don't think the credentials-per-namespace model of #1206 worked well for them either though - their model was to have multiple clusters in one namespace that could each be tagged with different credentials. I'm not sure how we'd map that into ASO: 
* By referring to credentials directly on ASO resources? 
* Some sub-namespace grouping of resources? 
* Resource group? That might be too restrictive.

Possibly it would make more sense to support their model directly without the rest of the credential fallback in #1206? (Although that would still require the code changes in ASO to make credential lookup per-request rather than when the reconciler is created.)